ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/util/BlockProcessorAltair.java

```diff
@@ -49,4 +69,83 @@ public void processAttestationsNoValidation(
+            state, beaconStateAccessors.getCurrentEpoch(state));
+    final SszVector<SszPublicKey> committeePubkeys = state.getCurrentSyncCommittee().getPubkeys();
+    final List<Integer> includedIndices = new ArrayList<>();
+    final List<BLSPublicKey> includedPubkeys = new ArrayList<>();
+    aggregate
+        .getSyncCommitteeBits()
+        .streamAllSetBits()
+        .forEach(
+            index -> {
+              includedIndices.add(committeeIndices.get(index));
+              includedPubkeys.add(committeePubkeys.get(index).getBLSPublicKey());
+            });
+    final Bytes32 domain =
+        beaconStateUtil.getDomain(
+            state,
+            altairSpecConfig.getDomainSyncCommittee(),
+            miscHelpers.computeEpochAtSlot(previousSlot));
+    final Bytes32 signingRoot =
+        beaconStateUtil.computeSigningRoot(
+            beaconStateUtil.getBlockRootAtSlot(state, previousSlot), domain);
+
+    if (!eth2FastAggregateVerify(
+        includedPubkeys, signingRoot, aggregate.getSyncCommitteeSignature().getSignature())) {
+      throw new BlockProcessingException("Invalid sync committee signature in " + aggregate);
+    }
+
+    // Compute the maximum sync rewards for the slot
+    final UInt64 totalActiveIncrements =
+        beaconStateAccessors
+            .getTotalActiveBalance(state)
+            .dividedBy(specConfig.getEffectiveBalanceIncrement());
+    final UInt64 totalBaseRewards =
+        altairBeaconStateAccessors.getBaseRewardPerIncrement(state).times(totalActiveIncrements);
+    final UInt64 maxEpochRewards =
+        totalBaseRewards
+            .times(IncentivizationWeights.SYNC_REWARD_WEIGHT)
+            .dividedBy(IncentivizationWeights.WEIGHT_DENOMINATOR);
+    final UInt64 maxSlotRewards =
+        maxEpochRewards
+            .times(includedIndices.size())
+            .dividedBy(committeeIndices.size())
+            .dividedBy(specConfig.getSlotsPerEpoch());
+
+    // Compute the participant and proposer sync rewards
+    final SszMutableList<Validator> validators = state.getValidators();
+    final UInt64 committeeEffectiveBalance =
+        includedIndices.stream()
+            .map(includedIndex -> validators.get(includedIndex).getEffective_balance())
+            .reduce(UInt64.ZERO, UInt64::plus)
+            .max(specConfig.getEffectiveBalanceIncrement());
+
+    final int beaconProposerIndex = beaconStateUtil.getBeaconProposerIndex(state);
+    includedIndices.forEach(
+        includedIndex -> {
+          final UInt64 effectiveBalance = validators.get(includedIndex).getEffective_balance();
+          final UInt64 inclusionReward =
+              maxSlotRewards.times(effectiveBalance).dividedBy(committeeEffectiveBalance);
+          final UInt64 proposerReward =
+              inclusionReward.dividedBy(specConfig.getProposerRewardQuotient());
+          validatorsUtil.increaseBalance(state, beaconProposerIndex, proposerReward);
```

Comment:
(nit) noticed potential micro-optimization: accumulate `proposerReward` and then do `increaseBalance` once, though not sure it worth deviating from the spec

Reply:
Seems like a good idea. There are up to 1024 sync committee participants so it adds up and block processing is an important bottleneck.

We'll need to do a proper optimisation pass through the Altair changes at some point as well.